### PR TITLE
In file-encryptor sample, remove padding after decrpytion on last block.

### PR DESCRIPTION
**Pull request by RRathna**
Also fixed requested_read_size assignment.

### CODE REVIEW

**Comment by RRathna** [comment]
Fix #3851

**Review by mingweishih — APPROVED** [review]

File: samples/file-encryptor/host/host.cpp [hunk 1]
```diff
@@ -266,9 +270,18 @@ int encrypt_file(
...
+        bytes_to_write = bytes_read;
+        // The data size is always padded to align with DATA_BLOCK_SIZE during encryption.
+        // Therefore, remove the padding (if any) from the last block during decryption.
+        if (!encrypt && bytes_left <= DATA_BLOCK_SIZE)
+        {
+            bytes_to_write = header.file_data_size % DATA_BLOCK_SIZE;
+            bytes_to_write = bytes_to_write > 0 ? bytes_to_write : bytes_read;
+        }
+
         if ((bytes_written = fwrite(
```
Comment (manojrupireddy) [inline samples/file-encryptor/host/host.cpp:282]: by removing equal sign in 276 line can we remove line 279 completely?

**Review by manojrupireddy — APPROVED** [review]

**Pull request merged**